chrome/content/zotero/elements/itemPane.js

```diff
@@ -0,0 +1,522 @@
+				this._viewType = type;
+			}
+		}
+
+		get pinnedPane() {
+			return this.getAttribute('pinnedPane');
+		}
+		
+		set pinnedPane(val) {
+			if (!val || !this.getPane(val)) {
+				val = '';
+			}
+			this.setAttribute('pinnedPane', val);
+			if (val) {
+				this._pinnedPaneMinScrollHeight = this._getMinScrollHeightForPane(this.getPane(val));
+			}
+			this.sidenav.updatePaneStatus(val);
+		}
+
+		get _minScrollHeight() {
+			return parseFloat(this._paneParent.style.getPropertyValue('--min-scroll-height') || 0);
+		}
+		
+		set _minScrollHeight(val) {
+			this._paneParent.style.setProperty('--min-scroll-height', val + 'px');
+		}
+
+		get sidenav() {
+			return this._sidenav;
+		}
+
+		set sidenav(sidenav) {
+			this._sidenav = sidenav;
+		}
+
+		static get observedAttributes() {
+			return ['pinnedPane'];
+		}
+
+		init() {
+			this._container = this.querySelector('#zotero-view-item-container');
+			this._header = this.querySelector('#zotero-item-pane-header');
+			this._paneParent = this.querySelector('#zotero-view-item');
+
+			this._container.addEventListener("keypress", this._handleKeypress);
+			this._paneParent.addEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb = new MutationObserver(this._handlePaneStatus);
+			this._paneHiddenOb.observe(this._paneParent, {
+				attributes: true,
+				attributeFilter: ["hidden"],
+				subtree: true,
+			});
+			this._initIntersectionObserver();
+
+			this._unregisterID = Zotero.Notifier.registerObserver(this, ['item'], 'itemPane');
+
+			this._disableScrollHandler = false;
+			this._pinnedPaneMinScrollHeight = 0;
+		}
+
+		destroy() {
+			this._container.removeEventListener("keypress", this._handleKeypress);
+			this._paneParent.removeEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb.disconnect();
+			this._intersectionOb.disconnect();
+
+			Zotero.Notifier.unregisterObserver(this._unregisterID);
+		}
+
+		async render() {
+			if (!this.initialized || !this.item) {
+				return;
+			}
+			let item = this.item;
+			Zotero.debug('Viewing item');
+			this._isRendering = true;
+
+			let panes = this.getPanes();
+			let pendingBoxes = [];
+			let inTrash = ZoteroPane.collectionsView.selectedTreeRow && ZoteroPane.collectionsView.selectedTreeRow.isTrash();
+			for (let box of [this._header, ...panes]) {
+				if (!box.showInFeeds && item.isFeedItem) {
+					box.style.display = 'none';
+					box.hidden = true;
+					continue;
+				}
+				else {
+					box.style.display = '';
+					box.hidden = false;
+				}
+				
+				if (this.mode) {
+					box.mode = this.mode;
+					
+					if (box.mode == 'view') {
+						box.hideEmptyFields = true;
+					}
+				}
+				else {
+					box.mode = 'edit';
+				}
+				
+				box.item = item;
+				box.inTrash = inTrash;
+				// Render sync boxes immediately
+				if (!box.hidden && box.render) {
+					if (box.render instanceof AsyncFunction) {
+						pendingBoxes.push(box);
```

Comment:
During item pane rendering, only sync `render` functions are called immediately and all async `render` functions wait until the scroll to section finishes. This fixes bugs with wrong scrolling positions caused by changing box height during scrolling.